### PR TITLE
Fix macOS ActivityIndicator to color properly in different themes

### PR DESCRIPTION
**Pull request by tom-un**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and Microsoft/react-native :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into Microsoft/react-native :thumbsup:
- [X] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

#### Description of changes

PROBLEM
ISS#3218502: LPC in Dark Mode: progress spinner doesn't spin/animate
The RCTActivityIndicatorView.m was flattening an NSColor and using its components to create layer contentFilter at the time that the `color` prop was being set.   In Mojave, NSColors should only ever be converted to colorspaces in one of four methods: updateLayer, drawRect:, layout, updateConstraints.  Since it was not performing the operation in any of these methods, the contentFilter was not responding to theme changes.

FIX
Override the updateLayer method and there flatten the _color prop to its components only if it exists, and then set the contentFilter.

PRODUCT IMPACT
Impacts any macOS SDX that uses the ActivityIndicator (LPC) and makes it render properly in all themes.

VERIFICATION 
Added semantic and dynamic colors in the ActivityIndicatorExample in RNTester to validate.

@markavitale ; @HeyImChris 

/azp run

### CODE REVIEW

File: React/Views/RCTActivityIndicatorView.m [hunk 1]
```diff
@@ -66,18 +66,23 @@ - (void)setActivityIndicatorViewStyle:(UIActivityIndicatorViewStyle)activityIndi
   }
 }
```
Comment (markavitale) [inline React/Views/RCTActivityIndicatorView.m]: Don't we still need a setting of the color property to result in kicking off a layer update? I think this is right, but we should double check that `setNeedsDisplay:` results in `updateLayer` being called on both iOS and macOS.
```suggestion
- (void)setColor: (UIColor*)color
{
  if (_color != color) {
    _color = color;
    [self setNeedsDisplay:YES];
  }
}
```

**Comment by tom-un** [comment]
It is in fact eslint that decided to make the ternary expression be a one liner in one case and a multi line in the other case – I assume because of length.

File: RNTester/js/ActivityIndicatorExample.js [hunk 2]
```diff
@@ -99,8 +100,18 @@ exports.examples = [
...
+          <ActivityIndicator
+            color={
+              Platform.OS === 'macos'
+                ? {dynamic: {light: 'black', dark: 'white'}}
+                : '#0000ff'
+            }
+          />
+          <ActivityIndicator
+            color={
+              Platform.OS === 'macos' ? {semantic: 'textColor'} : '#aa00aa'
```
[the remark this answers is not in the document]
Comment (tom-un) [reply RNTester/js/ActivityIndicatorExample.js]: Ya, eslint decided to make the short one a one linter and the longer one split on multiple lines.

File: React/Views/RCTActivityIndicatorView.m (hunk 1 above)
Comment (tom-un) [reply React/Views/RCTActivityIndicatorView.m]: Good idea. done.

**Comment by tom-un** [comment]
> Looks good so long as we've verified `updateLayer` is getting called correctly on both iOS and macOS.

It is on mac.  The code change does not effect ios

**Pull request merged**